broker.go

```diff
@@ -82,6 +84,38 @@ func (b *Broker) Open(conf *Config) error {
 		b.conn = newBufConn(b.conn)
 
 		b.conf = conf
+
+		if conf.Net.SASL.Enable {
+			//
+			// Begin SASL/PLAIN authentication
+			//
+			authBytes := []byte("\x00" + b.conf.Net.SASL.User + "\x00" + b.conf.Net.SASL.Password)
+			buf := new(bytes.Buffer)
```

Comment:
creating a new `bytes.Buffer` creates a growable dynamic buffer that resizes itself as you add more data. In this case it is unnecessary work, since you already know exactly how many bytes the message will be, so you can allocate the precise size up-front.